public/docs/ts/latest/guide/i18n.jade

```diff
@@ -0,0 +1,302 @@
+  
+
+a(id="extract")
+.l-main-section
+:marked
+  ## Extract and use translations
+  
+a(id="ng-xi18n")
+.l-main-section
+:marked
+  ### Extract messages with ng-xi18n
+  
+  Now that our templates have been updated to support i18n translations, we can use the `ng-xi18n` messages extractor.
+  This cli tool is based on `ngc`, and is available in the `@angular/compiler-cli` npm package.
+  
+  To use it, the first thing that we have to do is to install it:
+  
+code-example(language="sh").
+  npm install @angular/compiler-cli @angular/platform-server --save
+
+:marked
+  Like `ngc`, `ng-xi18n` is based on `tsc`, the TypeScript compiler.
+
+  Let's edit our `tsconfig.json` file and add the `angularCompilerOptions` with `genDir` pointing to the folder where we
+  want `ng-xi18n` to generate the file.
+
++makeExample('i18n/ts-snippets/i18n.config.snippets.ts', 'tsconfig')(format=".")
+
+:marked
+  We can then use the `ng-xi18n` binary to generate a file named `messages.xlf` at the root of our application.
+ 
+code-example(language="sh").
+  ./node_modules/.bin/ng-xi18n
+  
+.alert.is-helpful
+  :marked
+    It is considered good practice to create a new npm command that will be used to run `ng-xi18n`.
+    
+    Edit package.json and add the following command in the `scripts` property: `"extract": "ng-xi18n"`.
+    We can now generate our translations using the command `npm run extract`.
+
+:marked
+  If we want to generate this `messages.xlf` file somewhere in particular, `ng-xi18n` uses the same parameters as `ngc`.
+
+:marked
+  The generated `messages.xlf` file uses by default the format `XML Localisation Interchange File Format` (XLIFF, version 1.2).
+  
+  `ng-x18n` and Angular also support the `XML Message Bundle`(XMB) format. We can switch to this format by setting
+  the cli option `i18nFormat` to the value `xmb`:
+   
+code-example(language="sh").
+  ./node_modules/.bin/ng-xi18n --i18nFormat=xmb
+
+a(id="translate")
+.l-main-section
+:marked
+  ### Translate messages
+  
+  Now that we have generated a `messages` file, we have to make a copy by language that we want to support.
+  
+  If we want to support french for example, we can copy the file as `messages.fr`. We can start translating the messages,
+  or send those files to our translators.
+  
+  You can find a list of editors supporting the xlf format [here](https://en.wikipedia.org/wiki/XLIFF#Editors).
+  
+.alert.is-helpful
+  :marked
+    If we choose the default `XLIFF` format, we have to add the translations into the `<target/>` elements of our `messages.xlf`
+    file, such as: `<target>Bonjour i18n</target>`.
+    
+:marked
+  Whenever we add new messages in our application, we should run `ng-xi18n` again, and copy the added translations into
+  each of our localization files.
+  
+.alert.is-helpful
+  :marked
+    Using a versioning system such as `GIT` can help us find out easily what new translations have been generated by
```

Comment:
Removed advice to use `git`, replaced with `specialized translation software`. If you have an example of what could be used, I can add it in.